pkg/cache/tas_flavor_snapshot.go

```diff
@@ -296,6 +297,22 @@ func levelKey(topologyRequest *kueue.PodSetTopologyRequest) *string {
 	return nil
 }
 
+// findMostAllocatedDomainIdx binsearches a sorted array of domains and finds index of
+// the first domain with the lowest value of state, higher or equal than count.
+// If such a domain doesn't exist, it returns 0 as it's an index of the domain with the
+// most available resources
```

Comment:
could you align with the current impl? IIUC, this is not binary search.

Reply:
Done